apps/ui/lib/components/Event/Message/DisplayReactions.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import emoji from 'node-emoji';
+import styled from 'styled-components';
+import { useSelector } from 'react-redux';
+import { Button, Txt } from 'rendition';
+import { useSetup } from '../../SetupProvider';
+import { selectors } from '../../../store';
+import { Contract } from 'autumndb';
+
+const Emoji = ({ symbol, count, ...rest }) => {
+	const em = React.useMemo(() => {
+		return emoji.emojify(symbol);
+	}, [symbol]);
+
+	return (
+		<Txt {...rest}>
+			{em}
+			<span>{count}</span>
+		</Txt>
+	);
+};
+
+const Wrapper = styled.section`
+	display: flex;
+	flex-direction: row;
+	align-content: space-between;
+	width: 200px;
+`;
+
+export const DisplayReactions = ({ message }: { message: Contract }) => {
+	const currentUser = useSelector(selectors.getCurrentUser());
+
+	const reactions = message.links?.['has attached element']?.filter(
+		(contract) => {
+			return contract.type === 'reaction@1.0.0';
+		},
+	);
+
+	const [groupReactions, setGroupReactions] = useState(
+		{} as { [key: string]: Contract[] },
+	);
+
+	const { sdk } = useSetup()!;
+	// TODO: Remove this workaround when this is resolved
+	// https://github.com/product-os/autumndb/issues/1333
+	React.useEffect(() => {
+		if (!reactions) {
+			return;
+		}
+		const ac = new AbortController();
+		sdk
+			.query({
+				type: 'object',
+				$$links: {
+					'was created by': {
+						type: 'object',
+					},
+				},
+				properties: {
+					id: {
+						enum: reactions.map((reactionContract) => reactionContract.id),
+					},
+				},
+			})
+			.then((reactionsArray) => {
+				const result = reactionsArray.reduce((map, reactionContract) => {
+					const symbol = reactionContract.data.reaction as string;
+					map[symbol] = map[symbol] || [];
+					map[symbol].push(reactionContract);
+					return map;
+				}, {} as { [key: string]: Contract[] });
+				setGroupReactions(result);
+			})
+			.catch((err) => console.error(err));
+		return () => ac.abort();
+	}, [reactions]);
+
+	const handleClick = React.useCallback(
+		async (symbol) => {
+			const reactionByCurrentUser = groupReactions[symbol].find(
+				(reaction) =>
+					reaction.links?.['was created by']?.[0].id === currentUser['id'],
+			);
+			if (groupReactions[symbol] && reactionByCurrentUser) {
+				await sdk.card.remove(
+					reactionByCurrentUser.id,
+					reactionByCurrentUser.type,
+				);
+			} else {
+				const newReaction = await sdk.card.create({
+					type: 'reaction',
+					data: {
+						reaction: symbol,
+					},
+				});
+
+				await sdk.card.link(message, newReaction, 'has attached element');
+			}
+		},
+		[reactions, message],
```

Comment:
This callback does not have a dependency on `reactions` - you need to change this to `groupReactions`.
```suggestion
		[groupReactions, message],
```

Without this change, if you fix the infinite loop I commented on previously, then the onClick call back will not do anything, as `groupReactions` will always be the initial empty object.